### PR TITLE
Avoid empty cylinder description in git saving

### PR DESCRIPTION
**Pull request by atdotde**
The git parser complains with an error message when
reading an empty cylinder description string. So turn
the empty string into "unknown" upon saving.

Signed-off-by: Robert C. Helling <helling@atdotde.de>

### Describe the pull request:

- [x] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### CODE REVIEW

**Comment by atdotde** [comment]
OK, tests were not happy, maybe this is better.

**Comment by atdotde** [comment]
Not better. I will come back to this later tonight. Need to make dinner.

File: core/save-git.c [hunk 1]
```diff
@@ -144,6 +144,8 @@ static void save_cylinder_info(struct membuffer *b, struct dive *dive)
 		const char *description = cylinder->type.description;
 		int use = cylinder->cylinder_use;
 
+		if (description && !*description)
```
Comment (bstoeger) [inline core/save-git.c:147]: note: we have the `empty_string()` helper function.

**Comment by dirkhh** [comment]
The reason the tests fail is because we now change the data when written out. So you'll need to make appropriate adjustments to the test.

**Comment by bstoeger** [comment]
I'm sure that I'm missing something, but this appears a bit backwards to me. I see two possibilities:
1) We support empty cylinder descriptions. Then the git parser should not complain.
2) We do not support empty cylinder descriptions. Then this should be handled at a higher level. It would be weird to allow it in XML files but disallow it in git repositories, no?

**Comment by atdotde** [comment]
Forget this, this was fixing a bogon fluctuation bug

**Comment by atdotde** [comment]
It seems, the git parser is fine with having no description but it does not like 
description=""
I somehow managed to end up with the latter, though, but I cannot reproduce this anymore.

**Comment by bstoeger** [comment]
> I somehow managed to end up with the latter, though, but I cannot reproduce this anymore.

It's the same bug as #2788: Add a surface interval and you get a cylinder with empty description. The bug is already in 4.9.3... :(

**Pull request closed without merging**